backend/app/routers/feeds.py

```diff
@@ -50,9 +50,13 @@ async def check_feed_listeners(
         # Verify whether resource_id is found when searching the specified criteria
         feed_match = check_search_result(es_client, file_out, feed.search)
         if feed_match:
-            for listener in feed.listeners:
-                if listener.automatic:
-                    listener_ids_found.append(listener.listener_id)
+            feed_criteria = feed.search.criteria
+            for criterea in feed_criteria:
+                if criterea.field == 'content_type_main' and criterea.value == file_out.content_type.main_type:
```

Comment:
this code is redundant to what is already being called on line 51, `check_search_result`. content_type and content_type_main are in the elasticsearch index and will be applied to the search for feed_match, if we are inside this block of code this should already be confirmed.

Reply:
So for the query for JPEGS, here is the query string and here are the results:

`'{"preference":"results"}
{"query": {"bool": {"must": [{"match": {"name": "facebook_1633802270640_6852663398954784845.jpg"}}, {"bool": {"should": [{"match": {"content_type_main": "text"}}]}}]}}}

"{'took': 200, 'responses': [{'took': 200, 'timed_out': False, '_shards': {'total': 5, 'successful': 5, 'skipped': 0, 'failed': 0}, 'hits': {'total': {'value': 0, 'relation': 'eq'}, 'max_score': None, 'hits': []}, 'status': 200}]}"
'`

That is the correct query, but the conditional at the end always evaluated to True since it's checking the size of the entire response as a dictionary, and not whether the results had any hits. I have made another commit which should fix that.

I also sometimes got an error back from elasticsearch (timeout) and need to check if that comes up, or how this would be handled if it did.